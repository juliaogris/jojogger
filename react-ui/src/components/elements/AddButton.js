import React, { PropTypes } from 'react'
import Plus from '../../svgs/Plus'
const PlusButton = ({ onClick }) =>
  <div className='add-button-wrap'>
    <button className='add-button' onClick={onClick}>
      <Plus className='add-icon' />
    </button>
  </div>

PlusButton.PropTypes = {
  onClick: PropTypes.func.isRequired
}
export default PlusButton
