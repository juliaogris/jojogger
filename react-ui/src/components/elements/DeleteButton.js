import React from 'react'
import PropTypes from 'prop-types'

const DeleteButton = ({ show, onClick, item }) => {
  if (!show) {
    return null
  }
  return (
    <div className='delete-row'>
      <span className='delete-button' onClick={onClick}>
        Delete this {item}
      </span>
    </div>)
}
DeleteButton.PropTypes = {
  show: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  item: PropTypes.string.isRequired
}

export default DeleteButton
