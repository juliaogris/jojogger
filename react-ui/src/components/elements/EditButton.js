import React from 'react'
import PropTypes from 'prop-types'
import Pencil from '../../svgs/Pencil'
const EditButton = ({ onEdit, id }) => {
  return (
    <button onClick={() => onEdit(id)} className='edit-button'>
      <Pencil color={'#777'} />
    </button>
  )
}

EditButton.PropTypes = {
  onEdit: PropTypes.func.isRequired
}
export default EditButton
