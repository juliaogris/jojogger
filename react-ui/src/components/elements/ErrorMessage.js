import React, { PropTypes } from 'react'

const ErroMessage = ({ onCancel, error }) => {
  if (!error) {
    return null
  }
  const clickHandler = (event) => {
    event.preventDefault()
    onCancel(null)
  }
  return (
    <div className='error'>
      {error.message}
      <button onClick={clickHandler} className='error-cancel-button'>Ã—</button>
    </div>
  )
}
ErroMessage.PropTypes = {
  onCancel: PropTypes.func.isRequired,
  error: PropTypes.object.isRequired
}

export default ErroMessage
