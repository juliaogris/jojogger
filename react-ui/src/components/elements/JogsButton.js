import React from 'react'
import PropTypes from 'prop-types'
const JogsButton = ({ onJogsClick, id }) => {
  return (
    <button onClick={() => { onJogsClick(id) }}className='jogs-button'>
      Jogs
    </button>
  )
}

JogsButton.PropTypes = {
  onJogsClick: PropTypes.func.isRequired
}
export default JogsButton
