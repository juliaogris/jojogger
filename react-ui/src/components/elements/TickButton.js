import React, { PropTypes } from 'react'
import Tick from '../../svgs/Tick'
const TickButton = ({ onClick }) =>
  <div className='tick-button-wrap'>
    <button type='submit' className='tick-button' onClick={onClick}>
      <Tick color='white' className='add-icon' />
    </button>
  </div>

TickButton.PropTypes = {
  onClick: PropTypes.func.isRequired
}
export default TickButton
