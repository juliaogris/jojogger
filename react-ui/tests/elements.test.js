/* global it expect jest describe */
import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ReactTestUtils from 'react-addons-test-utils'
import AddButton from '../src/components/elements/AddButton'
import DeleteButton from '../src/components/elements/DeleteButton'
import EditButton from '../src/components/elements/EditButton'
import TickButton from '../src/components/elements/TickButton'
import JogsButton from '../src/components/elements/JogsButton'
import ErrorMessage from '../src/components/elements/ErrorMessage'

class Wrapper extends React.Component {
  render () {
    return this.props.children
  }
}

describe('AddButton', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<AddButton onClick={() => {}} />, div)
  })
  it('matches snapshot', () => {
    const tree = renderer.create(<AddButton onClick={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls "onClick()" on simulated click', () => {
    const onClick = jest.fn()
    const addButton = ReactTestUtils.renderIntoDocument(<Wrapper><AddButton onClick={onClick} /></Wrapper>)
    const buttons = ReactTestUtils.scryRenderedDOMComponentsWithTag(addButton, 'button')
    expect(buttons.length).toEqual(1)
    const button = buttons[0]
    ReactTestUtils.Simulate.click(button)
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})

describe('DeleteButton', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<DeleteButton show onClick={() => {}} item='MyItem' />, div)
  })
  it('matches snapshot', () => {
    const tree = renderer.create(<DeleteButton show onClick={() => {}} item='MyItem' />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('is null when show is set to false', () => {
    const tree = renderer.create(<DeleteButton show={false} onClick={() => {}} item='MyItem' />).toJSON()
    expect(tree).toBeNull()
  })
  it('calls "onClick()" on simulated click', () => {
    const onClick = jest.fn()
    const deleteButton = ReactTestUtils.renderIntoDocument(
      <Wrapper>
        <DeleteButton show onClick={onClick} item='MyItem' />
      </Wrapper>)
    const buttons = ReactTestUtils.scryRenderedDOMComponentsWithTag(deleteButton, 'span')
    expect(buttons.length).toEqual(1)
    const button = buttons[0]
    ReactTestUtils.Simulate.click(button)
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})

describe('EditButton', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<EditButton onEdit={() => {}} id={1} />, div)
  })
  it('matches snapshot', () => {
    const tree = renderer.create(<EditButton onEdit={() => {}} id={1} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls "onEdit()" on simulated click', () => {
    const onEdit = jest.fn()
    const editButton = ReactTestUtils.renderIntoDocument(<Wrapper><EditButton onEdit={onEdit} id={1} /></Wrapper>)
    const buttons = ReactTestUtils.scryRenderedDOMComponentsWithTag(editButton, 'button')
    expect(buttons.length).toEqual(1)
    const button = buttons[0]
    ReactTestUtils.Simulate.click(button)
    expect(onEdit).toHaveBeenCalledTimes(1)
    expect(onEdit).toHaveBeenCalledWith(1)
  })
})

describe('JogsButton', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<JogsButton onJogsClick={() => {}} />, div)
  })
  it('matches snapshot', () => {
    const tree = renderer.create(<JogsButton onJogsClick={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls "onJogsClick()" on simulated click', () => {
    const onJogsClick = jest.fn()
    const jogsButton = ReactTestUtils.renderIntoDocument(<Wrapper><JogsButton onJogsClick={onJogsClick} /></Wrapper>)
    const buttons = ReactTestUtils.scryRenderedDOMComponentsWithTag(jogsButton, 'button')
    expect(buttons.length).toEqual(1)
    const button = buttons[0]
    ReactTestUtils.Simulate.click(button)
    expect(onJogsClick).toHaveBeenCalledTimes(1)
  })
})

describe('TickButton', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<TickButton onClick={() => {}} />, div)
  })
  it('matches snapshot', () => {
    const tree = renderer.create(<TickButton onClick={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls "onClick()" on simulated click', () => {
    const onClick = jest.fn()
    const tickButton = ReactTestUtils.renderIntoDocument(<Wrapper><TickButton onClick={onClick} /></Wrapper>)
    const buttons = ReactTestUtils.scryRenderedDOMComponentsWithTag(tickButton, 'button')
    expect(buttons.length).toEqual(1)
    const button = buttons[0]
    ReactTestUtils.Simulate.click(button)
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})

describe('ErrorMessage', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ErrorMessage onCancel={() => {}} error={{ message: 'My Error' }} />, div)
  })
  it('matches snapshot', () => {
    const errorMessage = <ErrorMessage onCancel={() => {}} error={{ message: 'My Error' }} />
    const tree = renderer.create(errorMessage).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('is null when error is null', () => {
    const tree = renderer.create(<DeleteButton show={false} onClick={() => {}} />).toJSON()
    expect(tree).toBeNull()
  })

  it('calls "onCancel()" on simulated click', () => {
    const onCancel = jest.fn()
    const errorMessage = ReactTestUtils.renderIntoDocument(
      <Wrapper>
        <ErrorMessage onCancel={onCancel} error={{ message: 'My Error' }} />
      </Wrapper>)
    const buttons = ReactTestUtils.scryRenderedDOMComponentsWithTag(errorMessage, 'button')
    expect(buttons.length).toEqual(1)
    const button = buttons[0]
    ReactTestUtils.Simulate.click(button)
    expect(onCancel).toHaveBeenCalledTimes(1)
  })
})
