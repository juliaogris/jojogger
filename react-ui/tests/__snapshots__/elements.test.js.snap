exports[`AddButton matches snapshot 1`] = `
<div
  className="add-button-wrap">
  <button
    className="add-button"
    onClick={[Function]}>
    <svg
      className="add-icon"
      height="36px"
      viewBox="0 0 500 600"
      width="30px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink">
      <path
        d="M 241.99219,407.91016 L 241.99219,310.44922 L 144.72656,310.44922 L 144.72656,294.43359 L 241.99219,294.43359 L 241.99219,197.36328 L 257.61719,197.36328 L 257.61719,294.43359 L 355.27344,294.43359 L 355.27344,310.44922 L 257.61719,310.44922 L 257.61719,407.91016 L 241.99219,407.91016 z"
        fill="white" />
    </svg>
  </button>
</div>
`;

exports[`DeleteButton matches snapshot 1`] = `
<div
  className="delete-row">
  <span
    className="delete-button"
    onClick={[Function]}>
    Delete this 
    MyItem
  </span>
</div>
`;

exports[`EditButton matches snapshot 1`] = `
<button
  className="edit-button"
  onClick={[Function]}>
  <svg
    className=""
    height="14px"
    viewBox="0 0 600 600"
    width="14px"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink">
    <g
      fill="none"
      fillRule="evenodd"
      stroke="#777">
      <path
        d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
        strokeWidth="30" />
      <path
        d="m70.569 417.81 110.61 110.61"
        strokeWidth="25" />
      <path
        d="m491.47 108.37-366.69 366.68"
        strokeWidth="25" />
      <path
        d="m54.222 507.26 40.975 39.546"
        strokeWidth="25" />
    </g>
  </svg>
</button>
`;

exports[`ErrorMessage matches snapshot 1`] = `
<div
  className="page error">
  My Error
  <button
    className="error-cancel-button"
    onClick={[Function]}>
    Ã—
  </button>
</div>
`;

exports[`JogsButton matches snapshot 1`] = `
<button
  className="jogs-button"
  onClick={[Function]}>
  Jogs
</button>
`;

exports[`TickButton matches snapshot 1`] = `
<div
  className="tick-button-wrap">
  <button
    className="tick-button"
    onClick={[Function]}
    type="submit">
    <svg
      className="add-icon"
      height="100%"
      viewBox="0 -1  41 42"
      width="100%"
      xmlns="http://www.w3.org/2000/svg">
      <g
        fill="none"
        fillRule="evenodd"
        stroke="none"
        strokeWidth="1">
        <polygon
          fill="white"
          fillRule="nonzero"
          points="14.5299498 13.3700502 15.9299498 11.9700502 26.9458691 22.4458691 20.9458691 28.4458691 19.5458691 27.0458691 24.1458691 22.4458691"
          transform="translate(20.737909, 20.207960) rotate(90.000000) translate(-20.737909, -20.207960) " />
      </g>
    </svg>
  </button>
</div>
`;
